src/games/strategy/triplea/ai/proAI/util/ProMatches.java

```diff
@@ -374,6 +389,19 @@ public boolean match(final Territory t) {
     };
   }
 
+  public static Match<Territory> territoryHasNonMobileInfraFactoryAndIsNotConqueredOwnedLand(final PlayerID player,
```

Comment:
Any reason to not re-use  `territoryHasInfraFactoryAndIsNotConqueredOwnedLand`, ie: do an `and` match between `territoryHasInfraFactoryAndIsNotConqueredOwnedLand` and `territoryHasNonMobileInfraFactoryAndIsLand()`,